model_compression_toolkit/gptq/pytorch/gptq_graph_info.py

```diff
@@ -12,38 +12,60 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
+import torch
 import torch.nn as nn
 from typing import List
 from model_compression_toolkit.gptq.pytorch.quantizer.quantizer_wrapper import WeightQuantizerWrapper
 
 
 def get_trainable_parameters(fxp_model: nn.Module,
                              add_bias: bool = False,
-                             quantization_parameters_learning: bool = False) -> (List[nn.Parameter], List[nn.Parameter], List[nn.Parameter]):
+                             quantization_parameters_learning: bool = False,
+                             is_gumbel: bool = False) -> (List[nn.Parameter], List[nn.Parameter], List[nn.Parameter]):
     """
     Get trainable parameters from all layers in a model
 
     Args:
         fxp_model: Model to get its trainable parameters.
         add_bias: Whether to include biases of the model (if there are) or not.
         quantization_parameters_learning: Whether to include quantization parameters of the model or not.
+        is_gumbel: Whether the fxp model is quantized using Gumbel Rounding
     Returns:
         A list of trainable variables in a model. Each item is a list of a layers weights.
     """
 
     trainable_aux_weights = nn.ParameterList()
     trainable_threshold = nn.ParameterList()
     trainable_bias = nn.ParameterList()
+    trainable_temperature = nn.ParameterList()
 
     for layer in fxp_model.modules():
         if isinstance(layer, WeightQuantizerWrapper):
             trainable_aux_weights.append(layer.weight_quantizer.get_aux_variable())
             if quantization_parameters_learning:
                 trainable_threshold.append(layer.weight_quantizer.get_quantization_variable())
+            if is_gumbel:
+                trainable_temperature.append(layer.weight_quantizer.get_temperature_variable())
             if add_bias and layer.op.bias is not None:
                 trainable_bias.append(layer.op.bias)
 
-    return trainable_aux_weights, trainable_bias, trainable_threshold
+    return trainable_aux_weights, trainable_bias, trainable_threshold, trainable_temperature
+
+
+def get_gumbel_probability(fxp_model: nn.Module) -> List[torch.Tensor]:
```

Comment:
Why not use the same function as tf?

Reply:
It's not exactly the same. It uses framework stuff.